### ISSUE TITLE
font-subset has invalid signature on macOS aarch64

### ISSUE THREAD

**Issue by dg76** [flutter/flutter#129904]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Install Flutter 3.10.5 stable.
2. Create a new project `flutter create test`
3. Build the project `cd test; flutter build web`
4. The error `"font-subset" was blocked from use because it is not from an identified developer` appears and compilation fails:

```
Target web_release_bundle failed: IconTreeShakerException: Font subsetting failed with exit code -9.
```

As a workaround one can open macOS settings and click on "Security > Allow anyway". Then compile again and select "Open".

### Expected results

It should compile and "font-subset" should have a valid signature.

### Actual results

It does not compile.

### Code sample

A code sample can be created using `flutter create test`.

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[!] Flutter (Channel stable, 3.10.5, on macOS 12.6 21G115 darwin-arm64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] Android Studio (version 2021.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.2.3)
[✓] IntelliJ IDEA Community Edition (version 2022.2.3)
[✓] VS Code (version 1.76.0)
[✓] Connected device (5 available)
[✓] Network resources
```

</details>

**Comment by darshankawar** [comment]
@dg76 
A similar / related issue was fixed and merged in master channel recently. https://github.com/flutter/flutter/pull/128302
Can you switch to latest master and try again to see if the error still persists or not ?

`flutter channel master`
`flutter upgrade`

**Comment by dg76** [comment]
I cannot reproduce it any more. 🤔 I have installed Flutter 3.10.5 into a macOS Ventura VM to reproduce the problem but there the problem does not occur. Then I tried to execute `sudo spctl --reset-default` on my macOS Monterey computer where the problem had happened. Afterward I tried to build the project again but was not asked for any permissions. I don't know why the problem does not occur on Ventura and why it doesn't occur again after resetting the permissions on my Monterey system, sorry! Maybe it affects only Monterey and maybe resetting the permissions just didn't work. But it was there! 🙂 Anyway I will close this issue now.
